core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -295,6 +296,7 @@ public String toString() {
     this.previousFiles = previousFiles;
 
     this.snapshotsById = indexAndValidateSnapshots(snapshots, lastSequenceNumber);
+    this.snapshotTags = indexAndValidateSnapshotTags(snapshots);
```

Comment:
We do two passes to populate our snapshot structures now, maybe it makes sense to combine the new indexing operations into the same function?

Reply:
yes good point, let me update the function